storagenode/gracefulexit/worker.go

```diff
@@ -200,6 +207,61 @@ func (worker *Worker) transferPiece(ctx context.Context, transferPiece *pb.Trans
 	return c.Send(success)
 }
 
+// deletePieces delete pieces stored for a satellite
+// when no piece IDs are passed in, it will delete all pieces stored for the specified satellite.
+func (worker *Worker) deletePieces(ctx context.Context, pieceIDs []pb.PieceID) error {
+	// get piece size
+	pieceMap := make(map[pb.PieceID]int64)
+	ctxWithCancel, cancel := context.WithCancel(ctx)
+	err := worker.store.WalkSatellitePieces(ctxWithCancel, worker.satelliteID, func(piece pieces.StoredPieceAccess) error {
+		size, err := piece.ContentSize(ctx)
+		if err != nil {
+			worker.log.Debug("failed to retrieve piece info", zap.Stringer("Satellite ID", worker.satelliteID), zap.Error(err))
+		}
+		if pieceIDs == nil || len(pieceIDs) == 0 {
```

Comment:
nit - I'd feel better if it was just `if pieceIDs == nil {` in case someone builds an empty array.